sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -577,10 +577,10 @@ object SQLConf {
     .doc("The maximum lines of a single Java function generated by whole-stage codegen. " +
       "When the generated function exceeds this threshold, " +
       "the whole-stage codegen is deactivated for this subtree of the current query plan. " +
-      "The default value 2667 is the max length of byte code JIT supported " +
-      "for a single function(8000) divided by 3.")
+      "The default value 4000 is the max length of byte code JIT supported " +
+      "for a single function(8000) divided by 2.")
     .intConf
-    .createWithDefault(2667)
```

Comment:
I've already checked `2800`; this value activated too-long-function optimization in Q17/Q66 and Q66 had regression. So, I think `2731` and `2049` possibly have the same regression (I've also checked `2900` that disabled this in Q17/Q66). 

btw, why the computation is based on `8K`? It seems  the threshold of `DontCompileHugeMethods` is 8000?